### PR TITLE
Include OSGi JARs when generating AutoUpdate update.xml, otherwise HT…

### PR DESCRIPTION
**Pull request by jtulach**
…ML/Java APIs in 8.1 release may be omitted. I hope this is the right place to contribute patches now. Could this change be included in some new version of the plugin, hopefully available before 8.1 is out?

My application has dependency on 

```
    <dependency>
        <groupId>org.netbeans.html</groupId>
        <artifactId>ko-ws-tyrus</artifactId>
        <version>1.1</version>
    </dependency>
```

and its JARs are not included in the generated update catalog. My fix includes them.

### CODE REVIEW

**Comment by mkleint** [comment]
what versions of netbeans does this work with?

**Comment by jtulach** [comment]
I am trying to produce AU for an application based on NetBeans 8.1 Beta.

**Comment by mkleint** [comment]
ok, let me rephrase it, will this work with older versions of netbeans? eg. 8.0, 7.4.. or is the osgi on update center feature new to 8.1?

**Comment by jtulach** [comment]
OSGi bundles could be on AU center for a long time. It worked for trivial cases at least since 7.1 when we started to include Mylyn & etc. into NetBeans. There has been some bugfixes in 8.0 and now it seems to work reliably as documented by [Fair Minesweeper](http://plugins.netbeans.org/plugin/53864/fair-minesweeper) existence. Just nobody tried to generate the AU catalog with OSGi bundles via Maven, I guess.

**Comment by mkleint** [comment]
a problem with the solution:
the nbm-application project's project.getArtifacts() is used to get a list of artifacts including the transitive ones. On the fly we convert nbm project jars to nbm-file type (.nbm files) based on jar's manifest. That way we are able to pick out all module nbm files from transitive space. Unfortunately with  bundles (recognized by manifest entry as well) the same approach doesn't fly. A module jar can transitively pull a bundle in 2 ways: 
1. as something that is wrapped as part of nbm as plain jar on module's CP
2. as actual module dependency, depending on the bundle.

within getArtifacts() list there's no way to distinguish between 1 and 2. Your patch assumes case 2. But osgi dependencies in maven projects are actually turned OFF by default because most of the time people are not even aware that random artifacts from the internet have osgi headers (in some versions of the same artifact only in the worst cases) making the entire dependency management chaotic.  Please see CreateClusterAppMojo.java file for the logic how we recognize wrapped jars with osgi headers vs true osgi bundles.

**Comment by jtulach** [comment]
OK. what about explicitly saying I want my OSGi JARs in the catalog? With some plugin parameter. That would be compatible enough...

**Comment by mkleint** [comment]
a parameter that would work only in a subset of osgi related usecases?

**Comment by jtulach** [comment]
I've workarounded my current need by generating the update center differently. So we can probably close the request for now.
